source/brsHamcrest_Matchers/brsHamcrest_ObjectMatchers.brs

```diff
@@ -17,19 +17,33 @@ function sameObjectAs (value as Object) as Object
         value: value
 
         doMatch: function (target as Dynamic) as Boolean
+            result = false
             if (HasInterface(target, "ifAssociativeArray") AND HasInterface(m.value, "ifAssociativeArray"))
-                result = False
-                deviceInfo = CreateObject("roDeviceInfo")
-                uuidKey = "brsHamcrestUUID"
-                uuidValue = deviceInfo.GetRandomUUID()
-                target.AddReplace(uuidKey, uuidValue)
-                if (m.value.Lookup(uuidKey) = uuidValue) then result = True
-                target.Delete(uuidKey)
-                m.value.Delete(uuidKey)
-                return result
+                result = m._isSameObject(target, m.value)
+            else if (HasInterface(target, "ifArray") AND HasInterface(m.value, "ifArray"))
+                targetObj = {}
```

Comment:
I'm not sure if an empty associative array is unique enough to be safely considered the same object. I think having an array with an empty associative array as the final element isn't that unusual. Can we put some unique identifier inside?

Reply:
> I'm not sure if an empty associative array is unique enough to be safely considered the same object.

An empty AssocArray is still an object, and therefore we can check if another reference to an empty assocArrary is in fact the same reference. It is unique whether or not it contains any keys.

> I think having an array with an empty associative array as the final element isn't that unusual. Can we put some unique identifier inside?

It would not be possible to get any other array reference that had the same object at the same location. We make and destroy that object in the process of doing the match. The object added is tested in the same way we test AssocArrays in this match, with a temp UID.

Reply:
Ahhh, I read through the logic again, and you're right, I missed the part where `_isSameObject()` is used, and read it as comparing empty objects! My bad!